File: 56-MergeIntervals/56-MergeIntervals.cpp
// Last updated: 08/05/2026, 09:33:07
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        int n= intervals.size();
        int start;
        int end;
        vector<vector<int>> ans;
        sort(intervals.begin(), intervals.end());
        // Sort the vector based on the second element (index 1)
        for(int i=0; i<n;i++){
            if(i==0){
                start= intervals[i][0];
                end= intervals[i][1];
            }
            else if(end>=intervals[i][0]){
                end= max(intervals[i][1],end);
            }
            else{
                ans.push_back({start,end});
                start= intervals[i][0];
                end= intervals[i][1];
            }
        }
        ans.push_back({start,end});
        return ans;
        
    }
};

File: 56-MergeIntervals/56-MergeIntervals.py
# Last updated: 08/05/2026, 09:33:12
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        out = []
        for i in sorted(intervals, key=lambda i: i[0]):
            if out and i[0] <= out[-1][1]:
                out[-1][1] = max(out[-1][1], i[1])
            else:
                out += [i]
        return out
        
        
        

        
#         interval=[]
#         matched=[]
        
#         for n1,i in enumerate(intervals):
#             match=0
#             for n2,j in enumerate(intervals[n1+1:]):
                
#                 if i[1]>=j[0] and i[1]<=j[1] and i[0]>=j[0] and i[0]<=j[1]:
#                     matched.append(i)
#                 elif j[1]>=i[0] and j[1]<=i[1] and j[0]>=i[0] and j[0]<=i[1]:
#                     matched.append(j)
#                 elif i[0]==j[0] and i[1]==j[1]:
#                     pass
#                 elif i[1]>=j[0] and i[1]<=j[1]:
#                     add=[i[0],j[1]]
#                     interval.append(add)
#                     match=+1
#                     matched.append(i)
#                     matched.append(j)
#                 elif  i[0]<=j[1] and i[1]>=j[0]:
#                     add=[j[0],i[1]]
#                     interval.append(add)
#                     match=+1
#                     matched.append(i)
#                     matched.append(j)
            
#             if i not in matched and i not in interval:
#                 interval.append(i)
#         return interval